src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,562 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import styled from 'styled-components';
+import { AccountTypes, ExchangeOffer, NftList } from 'etherspot';
+import { TokenListToken } from 'etherspot/dist/sdk/assets/classes/token-list-token';
+import { ethers } from 'ethers';
+import debounce from 'debounce-promise';
+
+// Types
+import { IPlrDaoStakingMembershipBlock } from '../../types/transactionBlock';
+
+// Components
+import { useEtherspot, useTransactionBuilder } from '../../hooks';
+import Text from '../Text/Text';
+import AccountSwitchInput from '../AccountSwitchInput';
+import NetworkAssetSelectInput from '../NetworkAssetSelectInput';
+import TextInput from '../TextInput';
+import SelectInput, { SelectOption } from '../SelectInput/SelectInput';
+import { RoundedImage } from '../Image';
+
+// providers
+import { IAssetWithBalance } from '../../providers/EtherspotContextProvider';
+
+// utils
+import { formatAmountDisplay, formatMaxAmount } from '../../utils/common';
+import { addressesEqual, isValidEthereumAddress, isValidAmount } from '../../utils/validation';
+import { Chain, supportedChains, plrDaoAsset, plrDaoMemberNFT } from '../../utils/chain';
+import { swapServiceIdToDetails } from '../../utils/swap';
+
+//constants
+import { DestinationWalletEnum } from '../../enums/wallet.enum';
+
+export interface IPlrDaoTransactionBlockValues {
+  hasEnoughPLR: boolean;
+  fromChainId?: number;
+  fromAssetAddress?: string;
+  fromAssetIconUrl?: string;
+  fromAsset?: IAssetWithBalance;
+  selectedAsset?: IAssetWithBalance | null;
+  toAsset?: TokenListToken;
+  fromAssetDecimals?: number;
+  fromAssetSymbol?: string;
+  amount?: string;
+  accountType: AccountTypes;
+  receiverAddress?: string;
+  offer?: ExchangeOffer;
+}
```

Comment:
can we arrange this in a way so that mandatory types can come at the top and optional ones can come at the bottom?
Just for readability.